elastalert/elastalert.py

```diff
@@ -994,6 +1010,36 @@ def start(self):
             sleep_duration = total_seconds(next_run - datetime.datetime.utcnow())
             self.sleep_for(sleep_duration)
 
+    def wait_until_responsive(self, timeout, clock=timeit.default_timer):
+        """Wait until ElasticSearch becomes responsive (or too much time passes)."""
+
+        # Elapsed time is a floating point number of seconds.
+        timeout = timeout.total_seconds()
+
+        # Periodically poll ElasticSearch.  Keep going until ElasticSearch is
+        # responsive *and* the writeback index exists.
+        ref = clock()
+        while (clock() - ref) < timeout:
+            try:
+                if self.writeback_es.indices.exists(self.writeback_index):
```

Comment:
Yeah, that's intentional.

I understand that the typical use case involves setting up an ElasticSearch cluster, then running `elastalert-create-index`, but as described in the issue, I run `elastalert-create-index` after  `docker-compose up -d` that spawns both ElasticSearch & ElastAlert.

In order to make that work, I need this loop to wait until both these conditions are true:

1. ElasticSearch is responsive; and
2. The writeback index exists.